File: FLM_SubconLabelSystem/Library/Library.Database/BLL/PC1.vb
Namespace BLL
    ''' <summary>
    ''' Business Logic Layer
    ''' ---------------------------------
    ''' 18 Feb 2012   Yeon    Initial Version
    ''' </summary>
    ''' <remarks></remarks>
    Public Class PC1
        Inherits Library.Root.Other.BusinessLogicBase


        Public Shared Function List(ByVal Table As String, ByVal TableID As String, ByVal SearchField As String, ByVal SearchValue As String, ByVal SortField As String, ByVal Direction As Integer, _
                                    ByVal Page As Integer, ByVal Deleted As Integer) As ListCollection
            Using _dal As New DAL.PC1
                'Validation the parameter value                
                If Direction <> 1 Then
                    Direction = 0
                End If
                List = _dal.List(Table, TableID, SearchField, SearchValue, SortField, Direction, FromRowNo(Page), ToRowNo(Page), Deleted)
            End Using
        End Function

        Public Shared Function List2(ByVal refno As String, ByVal Table As String, ByVal TableID As String, ByVal SearchField As String, ByVal SearchValue As String, ByVal SortField As String, ByVal Direction As Integer, _
                            ByVal Page As Integer, ByVal Deleted As Integer) As ListCollection
            Using _dal As New DAL.PC1
                'Validation the parameter value
                If Direction <> 1 Then
                    Direction = 0
                End If

                List2 = _dal.List2(refno, Table, TableID, SearchField, SearchValue, SortField, Direction, FromRowNo(Page), ToRowNo(Page), Deleted)
            End Using
        End Function

        Public Shared Function GetData(ByVal ID As String) As DataTable
            Using _dal As New DAL.PC1
                GetData = _dal.GetData(ID)
            End Using
        End Function

        Public Shared Function GetDLLData(ByVal Value As String, ByVal ID As String) As DataTable
            Using _dal As New DAL.PC1
                GetDLLData = _dal.GetDLLData(Value, ID)
            End Using
        End Function

        Public Shared Function Maint(ByVal ID As String, ByVal PC1 As String, ByVal CompanyCode As String, ByVal NameDelivery As String, _
                                     ByVal RecType As String) As String
            Using _Dal As New DAL.PC1
                Dim str As String = System.Web.HttpContext.Current.Session("gstrUserID").ToString
                'Dim cc As String = System.Web.HttpContext.Current.Session("gstrUserCompCode").ToString

                Maint = _Dal.Maint(ID, PC1, CompanyCode, NameDelivery, RecType, Str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString)

                If Maint = "1" Then
                    _Dal.Commit()
                Else
                    _Dal.Rollback()
                End If
            End Using
        End Function
    End Class
End Namespace

File: FLM_SubconLabelSystem/Library/Library.Database/BLL/PC1.cs
using System.Data;

namespace Library.Database.BLL
{
    /// <summary>
    /// Business Logic Layer
    /// ---------------------------------
    /// 18 Feb 2012   Yeon    Initial Version
    /// </summary>
    public class PC1 : Library.Root.Other.BusinessLogicBase
    {
        public static ListCollection List(string Table, string TableID, string SearchField, string SearchValue, string SortField, int Direction,
                                          int Page, int Deleted)
        {
            using (var _dal = new DAL.PC1())
            {
                if (Direction != 1)
                {
                    Direction = 0;
                }
                return _dal.List(Table, TableID, SearchField, SearchValue, SortField, Direction, FromRowNo(Page), ToRowNo(Page), Deleted);
            }
        }

        public static ListCollection List2(string refno, string Table, string TableID, string SearchField, string SearchValue, string SortField, int Direction,
                                           int Page, int Deleted)
        {
            using (var _dal = new DAL.PC1())
            {
                if (Direction != 1)
                {
                    Direction = 0;
                }
                return _dal.List2(refno, Table, TableID, SearchField, SearchValue, SortField, Direction, FromRowNo(Page), ToRowNo(Page), Deleted);
            }
        }

        public static DataTable GetData(string ID)
        {
            using (var _dal = new DAL.PC1())
            {
                return _dal.GetData(ID);
            }
        }

        public static DataTable GetDLLData(string Value, string ID)
        {
            using (var _dal = new DAL.PC1())
            {
                return _dal.GetDLLData(Value, ID);
            }
        }

        public static string Maint(string ID, string PC1, string CompanyCode, string NameDelivery, string RecType)
        {
            using (var _Dal = new DAL.PC1())
            {
                string str = System.Web.HttpContext.Current.Session["gstrUserID"].ToString();
                string result = _Dal.Maint(ID, PC1, CompanyCode, NameDelivery, RecType, str, System.Web.HttpContext.Current.Request.UserHostAddress.ToString());

                if (result == "1")
                {
                    _Dal.Commit();
                }
                else
                {
                    _Dal.Rollback();
                }
                return result;
            }
        }

        public static string Maint(string ID, string PC1, string CompanyCode, string NameDelivery, string RecType, string userId, string userIp)
        {
            using (var _Dal = new DAL.PC1())
            {
                string result = _Dal.Maint(ID, PC1, CompanyCode, NameDelivery, RecType, userId, userIp);

                if (result == "1")
                {
                    _Dal.Commit();
                }
                else
                {
                    _Dal.Rollback();
                }
                return result;
            }
        }
    }
}
